pkg/cmd/roachtest/tests/kv.go

```diff
@@ -896,3 +897,130 @@ func registerKVRangeLookups(r registry.Registry) {
 		})
 	}
 }
+
+// measureQPS will measure the approx QPS at the time this command is run. The
+// duration is how long of an interval to wait while measuring. Setting too
+// short of an interval can mean inaccuracy in results. Setting too long of an
+// interval may mean the impact is blurred out.
+func measureQPS(ctx context.Context, t test.Test, db *gosql.DB, duration time.Duration) float64 {
+	numInserts := func() float64 {
+		var v float64
+		if err := db.QueryRowContext(
+			ctx, `SELECT value FROM crdb_internal.node_metrics WHERE name = 'sql.insert.count'`,
+		).Scan(&v); err != nil {
+			t.Fatal(err)
+		}
+		return v
+	}
+
+	before := numInserts()
+	time.Sleep(duration)
+	after := numInserts()
+	return (after - before) / duration.Seconds()
+}
+
+// registerKVRestartImpact measures the impact of stopping and then restarting a
+// node during a write-heavy workload. Specifically the Raft log on the node
+// falls behind when the node is down and when it comes back up it goes into IO
+// Overload as it attempts to recover. Note that this test stops the replicate
+// queue during the test to help isolate the impact of Raft backlog vs snapshot
+// transfers.
+func registerKVRestartImpact(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Skip:    "#95159",
+		Name:    "kv/restart/nodes=12",
```

Comment:
This is a relatively large ($$$) test with 12 nodes running for 30m. Could we make it smaller? We're using 8KiB writes below with only 50% of the total ops being just writes, but we could use larger writes or more blocks written per write, or a higher percentage of reads.